src/Controllers/cli_controller.py

```diff
@@ -26,11 +30,161 @@ def parse_list_cmd(self, d:UML_Diagram, tokens:list[str]):
+
+        Returns:
+            dict[str, set[str]]: A dictionary where keys are class names (str) and values are sets
+                                containing names of classes (str) with existing relations to be deleted.
+        """
+        res = dict()
+        for uml_class in self.get_classes():
+            existing_relations = set()
+            for rel in self._diagram.get_all_relations():
+                if rel.get_src_name() == uml_class:
+                    existing_relations.add(rel.get_dst_name())
+            if existing_relations:
+                res[uml_class] = existing_relations
+        return res
+
+    def get_add_relation(self) -> dict[str, dict[str, str]]:
+        """
+        Suggests class, class, relationship type 
+
+        Returns:
+        dict[str, dict[str, str]]: A dictionary where keys are class names (str)
+                                                 and values are dictionaries of class names (str) and
+                                                 corresponding relationship types (str).
+        """
+        rel_types = dict([
+            ("Aggregation", None),
+            ("Composition", None),
+            ("Realization", None),
+            ("Inheritance", None)
+        ])
+        class_reltype = dict((key, rel_types) for key in self.get_classes())
+        return dict((key, class_reltype) for key in self.get_classes())
+
+    def get_classes(self) -> dict[str, None]:
+        """
+        Get a dictionary of class names.
+
+        Returns:
+            dict[str, None]: An dictionary where keys are class names (str) and values are None.
+        """
+        return dict((c.get_name(), None) for c in self._diagram.get_all_classes())
+
+    def get_methods(self) -> dict[str, set[str]]:
+        """
+        Get a dictionary of methods for each class.
+
+        Returns:
+            dict[str, set[str]]: A dictionary where keys are class names (str) and
+                                        values are sets containing names of methods (str) for each class.
+        """
+        res = dict()
+        for uml_class in self._diagram.get_all_classes():
+            methods = set()
+            for method in uml_class.get_methods():
+                methods.add(method.get_name())
+            if methods:
+                res[uml_class.get_name()] = methods
+        return res
+
+    def get_fields(self) -> dict[str, set[str]]:
+        """
+        Get a dictionary of fields for each class.
+
+        Returns:
+            dict[str, set[str]]: A dictionary where keys are class names (str) and
+                                        values are sets containing names of fields (str) for each class.
+        """
+        res = dict()
+        for uml_class in self._diagram.get_all_classes():
+            fields = set()
+            for field in uml_class.get_fields():
+                fields.add(field.get_name())
+            if fields:
+                res[uml_class.get_name()] = fields
```

Comment:
this can be`` res[uml_class.get_name()] = set(uml_class.get_fields()) ``

Reply:
I need str values, which I'm getting with my current version

Reply:
right, this line of code doesn't create a set containing a list, it creates a set containing all the elements in the list. https://www.datacamp.com/tutorial/sets-in-python

Reply:
so the line of code above is saying "in the result map, assign a set of all the values in the fields list for the current class to the current class's name"